### PR TITLE
Add group view restrictions for events

### PR DESCRIPTION
**Pull request by iselinee**
Add backend for restricted view permissions on events.

### CODE REVIEW

**Review by LudvigHz — CHANGES_REQUESTED** [review]
Looks good!
Before we can merge this, we need to migrate all old events, so that they still work. We will probably need to create a custom migration to add `Abakom` to `can_view_groups` for all events that have `abakom_only=True`.

File: lego/apps/events/tests/test_events_api.py [hunk 1]
```diff
@@ -310,28 +334,6 @@ def test_creator_see_registrations(self):
         for pool in event_response.json()["pools"]:
             self.assertIsNotNone(pool.get("registrations", None))
 
-    def test_without_auth_permission_abakom_only(self):
```
Comment (erlingfn) [inline lego/apps/events/tests/test_events_api.py:313]: Did we have two copies of the same test?

**Review by erlingfn — APPROVED** [review]
Looks good! Great to get this fixed

**Review by eikhr — APPROVED** [review]

File: lego/apps/events/views.py [hunk 2]
```diff
@@ -307,6 +308,10 @@ def create(self, request, *args, **kwargs):
         serializer.is_valid(raise_exception=True)
         event_id = self.kwargs.get("event_pk", None)
         event = Event.objects.get(id=event_id)
+
+        if not get_permission_handler(Event).has_perm(request.user, VIEW, obj=event):
+            raise PermissionDenied()
```
Comment (ivarnakken) [inline lego/apps/events/views.py:313]: Good catch 💯

**Review by ivarnakken — APPROVED** [review]
Let's finally get this thing in ... 👴🏼

File: lego/apps/events/tests/test_events_api.py (hunk 1 above)
Comment (LudvigHz) [reply lego/apps/events/tests/test_events_api.py:313]: Looks like it yeah

**Review by LudvigHz — APPROVED** [review]

**Pull request merged**